src/Common/ShardChooser/ShardChooser.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\ShardChooser;
+
+final class ShardChooser
+{
+    private ?string $selectedShard;
+
+    /**
+     * @param string[] $shards
+     */
+    public function __construct(
+        private readonly Storage $storage,
+        private readonly array $shards
+    ) {
+        $this->selectedShard = null;
+    }
+
+    public function select(string $value): void
+    {
+        $shard = $this->shards[crc32($value) % count($this->shards)];
```

Comment:
This algorithm, although not deem necessary for this project, could be replaced by something more smart, e.g. [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing). Consistent hashing would reduce the likelihood of games being reassigned to another shard in the event of resharding.